### ISSUE TITLE
S3:put_public_access_block not adding values

### ISSUE THREAD

**Issue by sumeshsquare** [spulec/moto#2771]
#### Created S3 bucket and then triggered *put_public_access_block* as below:
s3.put_public_access_block(
Bucket=BucketName,
PublicAccessBlockConfiguration={'BlockPublicAcls': False,'IgnorePublicAcls': True,'BlockPublicPolicy': True,'RestrictPublicBuckets': True}
)

#### After that, when I tried calling below API, we were not getting the tags in response:

response = s3.get_public_access_block(
    Bucket=BucketName
)

**Response**:
{'ResponseMetadata': {'HTTPStatusCode': 200, 'HTTPHeaders': {}, 'RetryAttempts': 0}, 'PublicAccessBlockConfiguration': {}}

**Comment by bblommers** [comment]
Hi @sumeshsquare, there have been some improvements around public access blocks recently - can you reproduce this with the latest dev-release from moto? https://pypi.org/project/moto/#history

**Comment by sumeshsquare** [comment]
Hi @bblommers ,

Tested the same and the response have changed as expected:

```
Response:
{'ResponseMetadata': {'HTTPStatusCode': 200, 'HTTPHeaders': {}, 'RetryAttempts': 0}, 'PublicAccessBlockConfiguration': {'BlockPublicAcls': False, 'IgnorePublicAcls': True, 'BlockPublicPolicy': True, 'RestrictPublicBuckets': True}}
```

**Comment by SophieMJ** [comment]
Hi @bblommers , did the fix get released with version `1.3.16`? it seems like if I use `1.3.15.dev972` as mentioned in the comment above, I get the proper response, but if I use `1.3.16` which is the latest version I still get the empty response as noted in the original issue.

**Comment by bblommers** [comment]
Hi @SophieMJ, as far as I can see it actually got released with 1.3.15 already. This is the sample test case that I used to test this:
```
@mock_s3
def test_pab():
    bucket_name = "asdf"
    s3 = boto3.client("s3", region_name="us-east-1")
    s3.create_bucket(Bucket=bucket_name)
    s3.put_public_access_block(
        Bucket=bucket_name,
        PublicAccessBlockConfiguration={'BlockPublicAcls': False, 'IgnorePublicAcls': True, 'BlockPublicPolicy': True,
                                        'RestrictPublicBuckets': True}
    )
    response = s3.get_public_access_block(
        Bucket=bucket_name
    )
    print(response)
```
1.3.14: Empty PublicAccessBlockConfiguration-block
1.3.15: Returns everything
1.3.16: Returns everything
1.3.16.dev178: Returns everything

Are you able to share a reproducible test case, where you're getting an empty response?
